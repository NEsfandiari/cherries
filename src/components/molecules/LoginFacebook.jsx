import React, { Component } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { StyledButton } from '../atoms'
import { FaFacebookSquare } from 'react-icons/fa'
import FacebookLogin from 'react-facebook-login/dist/facebook-login-render-props'
import bcrypt from 'bcryptjs'
import postLambda from '../../utilities/postLambda'

const Container = styled.div`
  .facebook {
    display: flex;
    align-items: center;
    justify-content: center;
    svg {
      margin-left: 0.5rem;
    }
  }
`
class LoginFacebook extends Component {
  constructor(props) {
    super(props)
    this.responseFacebook = this.responseFacebook.bind(this)
  }

  // TODO: Handle error
  async responseFacebook(res) {
    let email = res.email
    let password = bcrypt.hashSync(
      `${res.id}${process.env.FACEBOOK_AUTH_KEYWORD}`,
      8
    )
    let lambdaResponse = await postLambda('getAccount', {
      email,
      password,
      remember: true,
    })
  }

  render() {
    return (
      <Container>
        <FacebookLogin
          appId={process.env.FACEBOOK_APP_ID}
          autoLoad
          fields="name,email,id"
          callback={this.responseFacebook}
          render={renderProps => (
            <StyledButton
              onClick={renderProps.onClick}
              width="17rem"
              height="2.5rem"
              color="#3B539A"
              backgroundColor="white"
              borderColor="#3B539A"
              borderWidth="2px"
              margin=".5rem"
              hovercolor="#E9F6FF"
              letterSpacing="0"
              className="facebook"
            >
              CONNECT WITH <FaFacebookSquare color="#3B539A" size="1.3rem" />
            </StyledButton>
          )}
        />
      </Container>
    )
  }
}

export default LoginFacebook

// OLD FACEBOOK LOGIN
//
//
// class LoginFacebook extends Component {
//   static contextTypes = {
//     firebase: PropTypes.object,
//   }

//   handleFacebook = e => {
//     const { firebase } = this.context
//     firebase.login(this, 'facebook')
//   }

//   render() {
//     return (
//       <Container>
//         <StyledButton
//           onClick={this.handleFacebook}
//           width="17rem"
//           height="2.5rem"
//           color="#3B539A"
//           backgroundColor="white"
//           borderColor="#3B539A"
//           borderWidth="2px"
//           margin=".5rem"
//           hovercolor="#E9F6FF"
//           letterSpacing="0"
//           className="facebook"
//         >
//           CONNECT WITH <FaFacebookSquare color="#3B539A" size="1.3rem" />
//         </StyledButton>
//       </Container>
//     )
//   }
// }
