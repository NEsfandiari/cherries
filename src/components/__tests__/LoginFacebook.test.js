import React from 'react'
import renderer from 'react-test-renderer'
import LoginFacebook from '../molecules/LoginFacebook'

describe('LoginFacebook', () =>
  it('renders correctly', () => {
    const tree = renderer.create(<LoginFacebook />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
