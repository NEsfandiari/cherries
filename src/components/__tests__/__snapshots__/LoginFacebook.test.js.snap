// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFacebook renders correctly 1`] = `
<div
  className="sc-jTzLTM jBzmmB"
>
  <button
    className="facebook sc-gzVnrw kzwOYl"
    color="#3B539A"
    height="2.5rem"
    letterSpacing="0"
    onClick={[Function]}
    width="17rem"
  >
    CONNECT WITH 
    <svg
      color="#3B539A"
      fill="currentColor"
      height="1.3rem"
      size="1.3rem"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "#3B539A",
        }
      }
      viewBox="0 0 448 512"
      width="1.3rem"
    >
      <path
        d="M448 80v352c0 26.5-21.5 48-48 48h-85.3V302.8h60.6l8.7-67.6h-69.3V192c0-19.6 5.4-32.9 33.5-32.9H384V98.7c-6.2-.8-27.4-2.7-52.2-2.7-51.6 0-87 31.5-87 89.4v49.9H184v67.6h60.9V480H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h352c26.5 0 48 21.5 48 48z"
      />
    </svg>
  </button>
</div>
`;
